lib/MongoDB/_Protocol.pm

```diff
@@ -420,9 +689,30 @@ sub parse_reply {
         }
         }
         if ( length($msg) < $len )
-        || ( $opcode != OP_REPLY )
+        || ( ( $opcode != OP_REPLY ) && ( $opcode != OP_MSG ) )
         || ( $response_to != $request_id );
 
+
+    if ( $opcode == OP_MSG ) {
+        MongoDB::InternalError->throw('OP_MSG requires codec for parse_reply') unless defined $codec;
+        # TODO Extract and check checksum
+        my @sections = split_sections( $codec, substr $msg, P_MSG_PREFIX_LENGTH );
+        # We have none of the other stuff? maybe flags... and an array of docs? erm
+        return {
+          flags => {
+            checksum_present => vec( $bitflags, MSG_FB_CHECKSUM, 1 ),
+            query_failure    => vec( $bitflags, MSG_FB_MORE_TO_COME, 1 ),
```

Comment:
More to come isn't the same thing as a query failure.  Have this be `more_to_come`.